### ISSUE TITLE
Typescript: InferSchemaType does not work as expected with ObjectId

### ISSUE THREAD

**Issue by hoangbn** [Automattic/mongoose#12594]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.6.7

### Node.js version

16.18.0

### MongoDB server version

5.x

### Typescript version (if applicable)

4.8.4

### Description

When `ObjectId` is a part of Schema, `InferSchemaType` does not produce expected result

### Steps to Reproduce

The following produces the type as expected
```
export const jobTemplateSchema = new Schema(
  {
    positionId: { type: Schema.Types.ObjectId, ref: "Position", required: true },
    description: { type: String, required: true },
  },
  {
    timestamps: true
  }
);

// The type
const jobTemplateSchema: Schema<any, Model<any, any, any, any, any>, {}, {}, {}, {}, "type", {
    positionId: Types.ObjectId;
    description: string;
}>
```

But when inferring it, it shows the following:

```
export type JobTemplateProps = InferSchemaType<typeof jobTemplateSchema>;

// The type
type JobTemplateProps = {
    positionId?: {
        _id?: ...;
        equals?: {};
        id?: {
            [x: number]: unknown;
            [Symbol.iterator]?: {};
            equals?: {};
            set?: {};
            toJSON?: {};
            find?: {};
            indexOf?: {};
            lastIndexOf?: {};
            slice?: {};
            length?: unknown;
            includes?: {};
            at?: {};
            buffer?: {
                slice?: {};
                [Symbol.toStringTag]?: unknown;
                byteLength?: unknown;
            } | {
                ...;
            };
            ... 91 more ...;
            valueOf: {};
        };
        ... 6 more ...;
        inspect?: {};
    };
    description?: unknown;
}
```

### Expected Behavior

The type should just be what we see in Schema type:
```
type JobTemplateProps = {
    positionId: Types.ObjectId;
    description: string;
}
```

**Comment by lpizzinidev** [comment]
Should be solved in mongoose `6.7.0` 💪🏻

**Comment by hoangbn** [comment]
Verified it's solved in `6.7.0`
